### ISSUE TITLE
Any plans to develop an OPCUA version?

### ISSUE THREAD

**Issue by nzfarmer1** [SoftwareTools4Makers/OPC#1]

**Comment by ildemartinez** [comment]
For Arduino UNO maybe could be a bit heavy but maybe for YÚN could be possible. Any plan to use it in a real project to test it?

All the best
ilde

**Comment by Martimeus** [comment]
Hi!
I think a OPCUA Version running on either the Yùn or perhaps the zero would be very interesting to test out. I can think of multiple uses for this at my workplace, ranging from valve automation to pressure control just to mention a few things. 

Regards, 
Martin

**Comment by ildemartinez** [comment]
Thank you Martin for the information and suggestions

At this moment I have no plans to integrate OPCUA because not enough time but maybe soon ;)

All the best
ilde

**Comment by dj-fiorex** [comment]
Sorry to reopen this! But if you want i can help you making an UA version of your great job! I'm working on a simulation platform based on a cortex m0+ (feather m0 wifi) and i want to integrate OPCUA architecture! 
I saw this project (https://github.com/Pro/open62541-arduino) but i can't build it with arduino ide!

**Comment by ildemartinez** [comment]
Hello

some Arduinos does not have enough power to server OPC UA....I am trying to do it with a Yun and OPC library from https://github.com/FreeOpcUa/python-opcua but need more space than the memory of the device....is a progress work

maybe could start from there?

all the best
ilde

**Comment by dj-fiorex** [comment]
Hello
Arduino Yun have both linux and arduino core, but my platform is based only on arduino core, meanwhile i wrote to the open62541-arduino repo to ask some help :)

**Comment by nzfarmer1** [comment]
You simply need to find an AT Mega based MCU that supports the hardware foot print necessary for the OPCUA stack and port accordingly.

**Comment by ProximaB** [comment]
I agree that this should be developed in OPC UA. At this moment this is working on COM/DCOM what is a huge drawback. There would be much more capabilities using Ethernet, you could use with phone etc. Then this project could become very useful.

**Comment by rony91jp** [comment]
I want to create an Arduino OPC UA server. Will it be possible to create an Arduino opc Ua server? Does Arduino equipped with OPC UA protocol?

**Comment by nzfarmer1** [comment]
Not enough SRAM.  You could start with an STM32 and it would be a worthy project.  Run it on a linux flavoured RTOS.

**Comment by rony91jp** [comment]
Thank you for your suggestion. Could you provide me any source that will help me? I want to communicate via LAN cable and want to send sensor analog data to the the OPC UA client.

**Comment by nzfarmer1** [comment]
You could start with something like this.  But I think you may be underestimating the effort a bit.

https://www.olimex.com/Products/IoT/ESP32/ESP32-POE/open-source-hardware

**Comment by dj-fiorex** [comment]
Hi all, sorry for my delay on this topic, i'm stuck with porting [Open62541](https://github.com/open62541/open62541) to my Feather M0 + WIZ5500 ethernet featherwing. 
I am not a cmake and code compilation expert so i have some trouble with this library!
If anyone has made progress in the meantime let us know!
